README.md

```diff
@@ -396,14 +396,71 @@ authClient.signIn({
 
 ### `signOut()`
 
-Signs the user out of their current Okta [session](https://developer.okta.com/docs/api/resources/sessions#example).
+Signs the user out of their current Okta [session](https://developer.okta.com/docs/api/resources/sessions#example) and clears all tokens stored in the `TokenManager`.
+
+By default:
+
+  - the user will be signed out using an API method 
+  - the page location will not be changed. 
+  
+When calling `signOut` with the default options, it is recommended to reload the page after the promise resolves to ensure all user data is cleared from memory and the app is properly loaded in an unauthenticated state.
+
+Since the API method relies upon an XHR call and the ability to set 3rd-party cookies, it may fail in certain scenarios, leaving the user's remote session open. For this reason, we generally recommend using the redirect method over the API method. The redirect method sets the window location to Okta where the session will be fully cleared.
+
+`signOut` takes the following options:
+
+* `redirect` - If `true`, enables the logout redirect method. By default, this will redirect the user to an Okta-hosted login page. Logging back in from this page may send the user to their Okta user homepage rather than the application.
+* `redirectUri` - Enables logout redirect using a custom URI to which the browser will be redirected after logout. This URI must be whitelisted as a `Logout Redirect Uri` in your application's general settings. It is common to use your application URI here.
+* `idToken` - Enables logout redirect and specifies the idToken. By default, `signOut` will look for a token named `idToken` within the `TokenManager`. If you have stored the idToken in a different location, you should retrieve it first and then pass it here. Passing a value of `false` for `idToken` will disable all logic with the `idToken`. In any case where the `idToken` is not available, the user will be signed out using the API method.
+
 
 ```javascript
+// Sign out using the API
 authClient.signOut()
 .then(function() {
   console.log('successfully logged out');
 })
-.fail(function(err) {
+.catch(function(err) {
+  console.error(err);
+})
+.then(function() {
+  // Reload to clear memory and see app in unauthenticated state
+  window.location.reload();
+});
+```
+
+
+```javascript
+// Sign out and redirect to Okta login
+authClient.signOut({ redirect: true })
+.then(function() {
+  console.log('successfully logged out');
```

Comment:
If we're doing a redirect, does this code ever run?

Reply:
Maybe. Probably not. The `catch()` block is more relevant. It could probably work if you somehow loaded your app into an iframe and were controlling it from a parent frame. Or were running inside some kind of shell like Electron where the script engine won't be unloaded by location change

Reply:
I removed `then` and `catch` block. We are catching all errors in this method, so there is no expectation of catching anything. I think I will leave the promise return for advanced cases and unit tests.